lib/matplotlib/cbook.py

```diff
@@ -2344,6 +2344,30 @@ def _picklable_class_constructor(mixin_class, fmt, attr_name, base_class):
     return cls.__new__(cls)
 
 
+def _is_torch_array(x):
+    """Check if 'x' is a PyTorch Tensor."""
+    try:
+        # we're intentionally not attempting to import torch. If somebody
+        # has created a torch array, torch should already be in sys.modules
+        return isinstance(x, sys.modules['torch'].Tensor)
+    except Exception:  # TypeError, KeyError, AttributeError, maybe others?
```

Comment:
I think this is too permissive, can we do this as 

```python
if sys.modules.get('torch', None) is not None:
     return isinstance(x, sys.modules['torch'].Tensor
```

and similar for jax?

Reply:
or

```python
if (mod := sys.modules.get('torch', None)) is not None:
    return isinstance(x, mod.Tensor)
```

Reply:
The failure mode there is that if someone names a file `jax.py` or `torch.py`, then matplotlib can no longer be used.

Reply:
fair point.  I retract my suggestion.